libsweep/include/sweep/sweep.h

```diff
@@ -10,12 +10,28 @@
 extern "C" {
 #endif
 
+#if defined _WIN32 || defined __CYGWIN__ || defined __MINGW32__
+// If we are building a dll, set SWEEP_API to export symbols
+#ifdef BUILDING_DLL
```

Comment:
Depending on whether you use the headers to create a dll or to consume a dll you need to prepend public functions with either `dllexport` or `dllimport`